src/modules/drop.ts

```diff
@@ -1122,52 +1173,229 @@ export class DropModule
+    if (!(await this.hasDelayedReveal())) {
+      throw new Error("delay reveal unsupported");
+    }
+
+    if (!password) {
+      throw new Error("Password is required");
+    }
+
+    const key = await this.hashDelayRevealPassword(batchId, password);
+
+    // performing the reveal locally to make sure it'd succeed before sending the transaction
+    try {
+      const decryptedUri = await this.contract.callStatic.reveal(batchId, key);
+
+      // basic sanity check for making sure decryptedUri is valid
+      // this is optional because invalid decryption key would result in non-utf8 bytes and
+      // ethers would throw when trying to decode it
+      if (!decryptedUri.includes("://") || !decryptedUri.endsWith("/")) {
+        throw new Error(
+          `Error revealing batch ${batchId} - make sure your password is correct`,
+        );
+      }
+    } catch (e) {
+      console.log(`Error revealing batch ${batchId}`, e);
+      throw new Error(
+        `Error revealing batch ${batchId} - make sure your password is correct`,
+      );
+    }
+
+    await this.sendTransaction("reveal", [batchId, key]);
+  }
+
+  /**
+   * Gets a list of token uris that needs to be revealed.
+   */
+  public async getBatchesToReveal(): Promise<BatchToReveal[]> {
+    if (!(await this.hasDelayedReveal())) {
+      throw new Error("Contract does not support delay reveal");
+    }
+
+    const count = await this.readOnlyContract.getBaseURICount();
+    if (count.isZero()) {
+      return [];
+    }
+
+    const countRangeArray = Array.from(Array(count.toNumber()).keys());
+
+    // map over to get the base uri indices, which should be the end token id of every batch
+    const uriIndices = await Promise.all(
+      countRangeArray.map((i) => this.readOnlyContract.baseURIIndices(i)),
+    );
+
+    // first batch always start from 0. don't need to fetch the last batch so pop it from the range array
+    const uriIndicesWithZeroStart = uriIndices.slice(0, uriIndices.length - 1);
+
+    // returns the token uri for each batches. first batch always starts from token id 0.
+    const tokenUris = await Promise.all(
+      Array.from([0, ...uriIndicesWithZeroStart]).map((i) =>
+        this.readOnlyContract.tokenURI(i),
+      ),
+    );
+
+    const tokenMetadatas = await Promise.all(
+      Array.from([0, ...uriIndicesWithZeroStart]).map((i) =>
+        this.getTokenMetadata(i.toString()),
+      ),
+    );
+
+    // index is the uri indicies, which is end token id. different from uris
+    const encryptedBaseUris = await Promise.all(
+      Array.from([...uriIndices]).map((i) =>
+        this.readOnlyContract.encryptedBaseURI(i),
+      ),
+    );
+
+    return tokenUris
+      .map((uri, index) => ({
+        batchId: BigNumber.from(index),
+        batchUri: uri,
+        placeholderMetadata: tokenMetadatas[index],
+      }))
+      .filter(
+        (_, index) => ethers.utils.hexDataLength(encryptedBaseUris[index]) > 0,
```

Comment:
oh you changed this? was there an issue with the way i had it?

Reply:
yeah because bytes32 get parsed to hex data in ethers (i believe)
so what that means for empty encryptedBaseUris -- instead of "", it actually returned "0x". 
instead of doing string check on hex bytes, it's probably a better to use the proper function to check the length of the bytes.